acceptance/tests/reconnect_after_broker_unavailable.rb

```diff
@@ -0,0 +1,38 @@
+require 'pxp-agent/config_helper.rb'
+require 'pxp-agent/test_helper.rb'
+
+test_name 'C94789 - An associated agent should automatically reconnect when the broker was temporarily unavailable'
+
+step 'Ensure each agent host has pxp-agent running and associated' do
+  agents.each_with_index do |agent, i|
+    on agent, puppet('resource service pxp-agent ensure=stopped')
+    cert_dir = configure_std_certs_on_host(agent)
+    create_remote_file(agent, pxp_agent_config_file(agent), pxp_config_json_using_test_certs(master, agent, i + 1, cert_dir).to_s)
+    on(agent, "rm -rf #{logfile(agent)}")
+    on agent, puppet('resource service pxp-agent ensure=running')
+    expect_file_on_host_to_contain(agent, logfile(agent), PXP_AGENT_LOG_ENTRY_ASSOCIATION_SUCCESS, 60)
+  end
+end
+
+step 'On master, stop the broker for at least 5 seconds' do
+  kill_pcp_broker(master)
+  sleep(5)
```

Comment:
Can the kill method be modified to make sure the process is killed rather than putting a random sleep here?

`kill -0 <pid>` or setting a sleep loop in the method with a timeout threshold?